pkg/api/componentreadiness/middleware/releasefallback/releasefallback.go

```diff
@@ -341,7 +342,31 @@ func newFallbackTestQueryReleasesGenerator(
 		ReqOptions:  reqOptions,
 	}
 	return generator
+}
 
+type fallbackTestQueryReleasesGeneratorCacheKey struct {
+	BaseRelease     string
+	BaseStart       time.Time
+	BaseEnd         time.Time
+	AdvancedOptions reqopts.Advanced
+	// VariantDBGroupBy is the only field within VariantOption that is used here
+	VariantDBGroupBy sets.String
+	// CRTimeRoundingFactor is used by GetReleaseDatesFromBigQuery
+	CRTimeRoundingFactor time.Duration
+}
+
+// getCacheKey creates a cache key using the generator properties that we want included for uniqueness in what
+// we cache. This provides a safer option than using the generator previously which carries some public fields
+// which would be serialized and thus cause unnecessary cache misses.
+func (f *fallbackTestQueryReleasesGenerator) getCacheKey() fallbackTestQueryReleasesGeneratorCacheKey {
+	return fallbackTestQueryReleasesGeneratorCacheKey{
+		BaseRelease:          f.BaseRelease,
+		BaseStart:            f.BaseStart,
+		BaseEnd:              f.BaseEnd,
+		AdvancedOptions:      f.ReqOptions.AdvancedOption,
```

Comment:
That is interesting.  Agree that we need to honor `AdvancedOption.IgnoreDisruption` in the cache for the fallback queries even though it isn't likely to change (doubt anyone does) but things like pity would cause the TestDetails(?) to take along time to load on a cache miss for fallback and not change the results.

Doubt that happens a lot but if you were confident that was the only thing used it could be nice to reuse the cache in a case like that.

Reply:
Yeah, I think it would make sense to only store `IgnoreDisruption` here